pkg/phlaredb/block_querier.go

```diff
@@ -905,6 +938,75 @@ func (b *singleBlockQuerier) SelectMatchingProfiles(ctx context.Context, params
 	return iter.NewMergeIterator(maxBlockProfile, false, iters...), nil
 }
 
+func (b *singleBlockQuerier) Series(ctx context.Context, params *ingestv1.SeriesRequest) ([]*typesv1.Labels, error) {
```

Comment:
The big question will be how accurate do we want to be. Without looking at the `profiles.parquet`, we will have a label resolution for the block length only. 

Let's say we only have profile for pod-a from t1 until t1+5min. Label queries will still show pod-a if we query t1+1h until t1+2, just because a block containing the is going from t1-1h until t1+2h.  

I do think this only a minor problem and would rather keep things simple and index only.

We should add a comment about this though.

Reply:
Yeah, I did think about that too. The block-length query resolution is unfortunate, but I think something we can iterate on.

Added a comment about this in: https://github.com/grafana/pyroscope/pull/2431/commits/cc05423aed7d23e234515c0a96de77ebea91756d